python-package/xgboost/testing/__init__.py

```diff
@@ -352,137 +352,6 @@ def __repr__(self) -> str:
         return self.name
 
 
-@memory.cache
```

Comment:
pylint complains about the file being too huge (>1000 loc). I moved some of the data fetchers into `testing/data.py`.